docs/userguide/serialization.md

```diff
@@ -0,0 +1,391 @@
+
+The following sections given an overview of the flexibilities that are offered for this
+task. Of course, the underlying concepts can be mixed and matched arbitrarily.
+
+### Basic string assembly
+Writing a configuration for a certain BayBE object in form of a serialization string is
+easy:
+1. Select your desired object class
+2. Identify the arguments expected by one of its constructors (see also [here](#ALTERNATIVE_CONSTRUCTORS))
+3. Pack them into a JSON string that mirrors the constructor signature
+
+Let's have a more detailed look, for instance, at the serialization string from
+the [above example](#JSON_SERIALIZATION), this time assuming we wanted to assemble
+the string manually.
+For this purpose, we have a peek at the signature of the `__init__` method of
+{meth}`CategoricalParameter <baybe.parameters.categorical.CategoricalParameter.__init__>`
+and notice that it has two required arguments, `name` and `values`.
+We specify these accordingly as separate fields in the JSON string:
+
+```python
+from baybe.parameters import CategoricalParameter
+
+parameter_str = """
+{
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+from_json = CategoricalParameter.from_json(parameter_str)
+via_init = CategoricalParameter(name="setting", values=["low", "high"])
+
+assert from_json == via_init
+```
+
+### Using default values
+Just like default values can be omitted when working in Python ...
+```python
+from baybe.parameters import CategoricalParameter
+
+p1 = CategoricalParameter(name="setting", values=["low", "high"])
+p2 = CategoricalParameter(name="setting", values=["low", "high"], encoding="OHE")
+
+assert p1 == p2
+```
+... they can be omitted from the corresponding serialization string.
+```python
+from baybe.parameters import CategoricalParameter
+
+p1_str = """
+{
+    "name": "setting",
+    "values": ["low", "high"]
+}
+"""
+p2_str = """
+{
+    "name": "setting",
+    "values": ["low", "high"],
+    "encoding": "OHE"
+}
+"""
+
+assert CategoricalParameter.from_json(p1_str) == CategoricalParameter.from_json(p2_str)
+```
+
+### Automatic field conversion
+You may have noticed that BayBE classes apply converters to their inputs so that
+simpler attribute representations can be passed.
+Of course, these shortcuts can be analogously used inside a configuration string.
+
+While the above holds generally true for all classes that have converters in place,
+providing a few specific example may help to convey the concept:
+
+* Since {class}`Intervals <baybe.utils.interval.Interval>` can be created _implicitly_,
+    it is enough the specify their bound values directly:
+```python
+from baybe.targets import NumericalTarget
+from baybe.utils.interval import Interval
+
+t1 = NumericalTarget(name="t", mode="MAX", bounds=Interval(0, 1))
+t2 = NumericalTarget(name="t", mode="MAX", bounds=(0, 1))
+t3 = NumericalTarget.from_json('{"name": "t", "mode": "MAX", "bounds": [0, 1]}')
+
+assert t1 == t2 == t3
```

Comment:
I actually disagree here:
* Removing the assert given our current setup pipeline is a no-go, because we need to ensure it's tested (we can talk about removing it in case we ever switch to the sphinx `doctest` plugin and run the assert behind the scenes – but for now it's needed)
* if you have an `assert` like this in the code, I don't see how this could NOT indicate to the users that the equality is supposed to hold
* quite to the contrary, putting a `print` instead would have zero meaning: neither would stuff be tested, nor would the users see the print result
* plus I think the assert makes even more sense in combination with the copy-code button we provide

So I'll resolve the thread, but feel free to reopen if you want to discuss further